ecs-cli/modules/cli/regcreds/regcreds_app_test.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2015-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package regcreds
+
+import (
+	"testing"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients/aws/secretsmanager/mock"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/utils/regcreds"
+	"github.com/aws/aws-sdk-go/aws"
+	secretsmanager "github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/golang/mock/gomock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetOrCreateRegistryCredentials_WithCredPair(t *testing.T) {
+
+	testUsername := "someUser1"
+	testPassword := "someP4$$w0rd"
+	testRegistryName := "examplereg.net"
+	testContainers := []string{"logging", "web"}
+
+	testRegistryCreds := make(map[string]readers.RegistryCredEntry)
+	testRegistryCreds[testRegistryName] = getTestCredsEntry("", testUsername, testPassword, "", testContainers)
+
+	expectedCreateInput := secretsmanager.CreateSecretInput{
+		Name:         generateSecretName(testRegistryName),
+		SecretString: generateSecretString(testUsername, testPassword),
+		Description:  generateSecretDescription(testRegistryName),
+	}
+	responseARN := "arn:aws:secretsmanager:examplereg.net-123"
+
+	mockSM := setupTestController(t)
+	gomock.InOrder(
+		mockSM.EXPECT().DescribeSecret(gomock.Any()).Return(nil, nil),
+		mockSM.EXPECT().CreateSecret(expectedCreateInput).Return(&secretsmanager.CreateSecretOutput{ARN: aws.String(responseARN)}, nil),
+	)
+
+	credsOutput, err := getOrCreateRegistryCredentials(testRegistryCreds, mockSM, false)
+	assert.NoError(t, err, "Expected no error when creating secret with cred pair")
+
+	credEntryMap := *credsOutput
```

Comment:
This is a bit odd, though I see it's bc `getOrCreateRegistryCredentials` returns a pointer to a map. Since that output seems to only be used for testing purposes rather than in the implementation itself, plus the fact that maps are already sort of pointers underneath the hood, could we just return the map itself? Then everywhere you do this in the tests, you can just delete `credEntryMap := *credsOutput` and simplify the following line to `actualCredEntry := credsOutput[testRegistryName]`

Reply:
The `getOrCreateRegistryCredentials` output will be used in later CRs for role & output creation. However, the main reason it's a pointer is so I don't have to return an empty `map[string]CredsOutputEntry` (vs. nil) in the event the function returns an error. Though I suppose I could initialize an empty map at the beginning and return it unmodified along w/ the error in that case.

Reply:
I second HHH's comment, having a pointer to a map is a bit unusual in Go